testing/tests/physics.lua

```diff
@@ -0,0 +1,685 @@
+  test:assertEquals(nil, endContact, 'check no end contact callback')
+  test:assertEquals(nil, preSolve, 'check no pre solve callback')
+  test:assertEquals(nil, postSolve, 'check no post solve callback')
+  local beginContactCheck = false
+  local endContactCheck = false
+  local preSolveCheck = false
+  local postSolveCheck = false
+  local collisions = 0
+  world:setCallbacks(
+    function() beginContactCheck = true; collisions = collisions + 1 end,
+    function() endContactCheck = true end,
+    function() preSolveCheck = true end,
+    function() postSolveCheck = true end
+  )
+  local body2 = love.physics.newBody(world, 10, 10, 'dynamic')
+  local rectangle2 = love.physics.newRectangleShape(body2, 0, 0, 10, 10)
+  test:assertEquals(false, beginContactCheck, 'check world didnt update after adding body')
+  world:update(1)
+  test:assertEquals(true, beginContactCheck, 'check contact start')
+  test:assertEquals(true, preSolveCheck, 'check pre solve')
+  test:assertEquals(true, postSolveCheck, 'check post solve')
+  body2:setPosition(100, 100)
+  world:update(1)
+  test:assertEquals(true, endContactCheck, 'check contact end')
+  -- check point checking
+  local shapes = 0
+  world:queryShapesInArea(0, 0, 10, 10, function(x)
+    shapes = shapes + 1
+  end)
+  test:assertEquals(1, shapes, 'check shapes in area')
+  world:rayCast(0, 0, 200, 200, function(x)
+    shapes = shapes + 1
+    return 1
+  end)
+  test:assertEquals(3, shapes, 'check shapes in raycast')
+  -- change collision logic
+  test:assertEquals(nil, world:getContactFilter(), 'check def filter')
+  world:update(1)
+  world:setContactFilter(function(s1, s2)
+    return false -- nothing collides
+  end)
+  body2:setPosition(10, 10)
+  world:update(1)
+  test:assertEquals(1, collisions, 'check collision logic change')
+  -- final bits
+  world:setGravity(1, 1)
+  test:assertEquals(1, world:getGravity(), 'check grav change')
+  world:destroy()
+  test:assertEquals(true, world:isDestroyed(), 'check world destroyed')
+end
+
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+------------------------------------METHODS-------------------------------------
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+
+-- love.physics.getDistance
+love.test.physics.getDistance = function(test)
+  -- setup two fixtues to check
+  local world = love.physics.newWorld(0, 0, false)
+  local body = love.physics.newBody(world, 10, 10, 'static')
+  local shape1 = love.physics.newEdgeShape(body, 0, 0, 5, 5)
+  local shape2 = love.physics.newEdgeShape(body, 10, 10, 15, 15)
+  -- check distance between them
+  test:assertEquals(647106, math.floor(love.physics.getDistance(shape1, shape2)*100000), 'check distance matches')
```

Comment:
Is there a sort of 'assert number equals with the given tolerance' helper function this could use, instead of having to apply ad hoc tolerance code?